### PR TITLE
Initial Commit for BeeHive

### PR DESCRIPTION
**Pull request by lidiyag**
Moved BeeHive from kubeedge to new repo for BeeHive

### CODE REVIEW

**Review by sids-b — APPROVED** [review]
LGTM. @m1093782566 should we add KubeEdge LICENSE head on all files from now itself ??

File: pkg/common/log/logger.go [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+		if value == "file" {
+			rotate.RunLogRotate(logConfig.LoggerFile, &rotate.RotateConfig{}, LOGGER)
+			break
+		}
+	}
+
+	LOGGER.Debug("init logger...")
+}
+
+// Trace 放在一个函数的开头，会打印整个函数的运行时间
```
Comment (m1093782566) [inline pkg/common/log/logger.go]: // Trace will output the execution time for a given function.

File: pkg/common/log/logger_test.go [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+import (
+	"testing"
+
+	"github.com/kubeedge/beehive/pkg/common/log"
+	//"os"
+)
+
+func TestLoggerInitilization(t *testing.T) {
+	log.LOGGER.Debug("testing log")
+	//_, err := os.Stat("edge.log")
```
Comment (m1093782566) [inline pkg/common/log/logger_test.go]: Please remove the deadlines.

File: pkg/common/log/logger_test.go [hunk 3]
```diff
@@ -0,0 +1,23 @@
+package log_test
+
+import (
+	"testing"
+
+	"github.com/kubeedge/beehive/pkg/common/log"
+	//"os"
```
Comment (m1093782566) [inline pkg/common/log/logger_test.go]: Please remove the deadline

File: pkg/common/log/logger_test.go [hunk 4]
```diff
@@ -0,0 +1,23 @@
+package log_test
```
Comment (m1093782566) [inline pkg/common/log/logger_test.go]: package log

File: pkg/common/util/conn_test.go [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+		fmt.Println(context)
+		now := "response from server: " + time.Now().String()
+		return now
+	})
+	go us.StartServer()
+	time.Sleep(time.Second * 10)
+}
+
+func TestClient(t *testing.T) {
+	//声明unixsocket
```
Comment (m1093782566) [inline pkg/common/util/conn_test.go]: Remove this comment as it's not meaningful.

File: test/modules/test_source.go [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+	if err != nil {
+		fmt.Printf("failed to send sync message, error:%v\n", err)
+	} else {
+		fmt.Printf("get resp: %v\n", resp)
+	}
+
+	message = model.NewMessage("").SetRoute(SourceModule, DestinationGroup).
+		SetResourceOperation("test", model.DeleteOperation).FillBody("fine")
+	c.Send2Group(DestinationGroup, *message)
+	//message = model.NewMessage("").SetRoute(SourceModule, DestinationGroup).
```
Comment (m1093782566) [inline test/modules/test_source.go]: Please remove the dead lines.

File: vendor/github.com/ServiceComb/go-archaius/.gitignore [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+
+# Test binary, build with `go test -c`
+*.test
+
+# Output of the go coverage tool, specifically when used with LiteIDE
+*.out
+
+# Project-local glide cache, RE: https://github.com/Masterminds/glide/issues/736
+.glide/
+.idea
```
Comment (m1093782566) [inline vendor/github.com/ServiceComb/go-archaius/.gitignore:15]: newline please.

File: vendor/github.com/ServiceComb/go-archaius/.travis.yml [hunk 8]
```diff
@@ -0,0 +1,42 @@
+language: go
```
Comment (m1093782566) [inline vendor/github.com/ServiceComb/go-archaius/.travis.yml:1]: I don't think file is needed.

**Comment by sids-b** [comment]
cc @qizha

File: vendor/github.com/ServiceComb/paas-lager/.travis.yml [hunk 9]
```diff
@@ -0,0 +1,36 @@
+language: go
```
Comment (m1093782566) [inline vendor/github.com/ServiceComb/paas-lager/.travis.yml:1]: I don't think file is needed.

**Review by m1093782566 — CHANGES_REQUESTED** [review]
Will be appreciated if you can split the vendor commit.

**Comment by m1093782566** [comment]
> @m1093782566 should we add KubeEdge LICENSE head on all files from now itself ??

For LICENSE, we need to decide if beehive will sit in kubeedge org in the future. If so, we can add the kubeedge LICENSE HEAD or we need to move beehive out of kubeedge and use its own LICENSE.

@qizha thoughts?

File: vendor/github.com/ServiceComb/go-archaius/.travis.yml (hunk 8 above)
Comment (lidiyag) [reply vendor/github.com/ServiceComb/go-archaius/.travis.yml:1]: @m1093782566 , we can remove the yaml file but a dep ensure will create it again..Please let me know whether this should be removed.

Comment (m1093782566) [reply vendor/github.com/ServiceComb/go-archaius/.travis.yml:1]: Okay, just keep this file.

**Comment by sids-b** [comment]
@lidiyag please resolve other comments of @m1093782566 so that we can merge this quickly.

File: vendor/github.com/ServiceComb/go-archaius/.gitignore (hunk 7 above)
Comment (lidiyag) [reply vendor/github.com/ServiceComb/go-archaius/.gitignore:15]: @m1093782566 the addition of new line also goes back when dep ensure is done. Please confirm whether I should add the same.

File: pkg/common/log/logger.go (hunk 1 above)
Comment (lidiyag) [reply pkg/common/log/logger.go]: Updated

File: pkg/common/log/logger_test.go (hunk 3 above)
Comment (lidiyag) [reply pkg/common/log/logger_test.go]: Updated

File: pkg/common/log/logger_test.go (hunk 4 above)
Comment (lidiyag) [reply pkg/common/log/logger_test.go]: Updated

File: pkg/common/util/conn_test.go (hunk 5 above)
Comment (lidiyag) [reply pkg/common/util/conn_test.go]: Updated

File: test/modules/test_source.go (hunk 6 above)
Comment (lidiyag) [reply test/modules/test_source.go]: Updated

File: vendor/github.com/ServiceComb/go-archaius/.travis.yml (hunk 8 above)
Comment (lidiyag) [reply vendor/github.com/ServiceComb/go-archaius/.travis.yml:1]: Updated

File: vendor/github.com/ServiceComb/paas-lager/.travis.yml (hunk 9 above)
Comment (lidiyag) [reply vendor/github.com/ServiceComb/paas-lager/.travis.yml:1]: Kept the file, as dep ensure brings it back

**Comment by qizha** [comment]
> > @m1093782566 should we add KubeEdge LICENSE head on all files from now itself ??
> 
> For LICENSE, we need to decide if beehive will sit in kubeedge org in the future. If so, we can add the kubeedge LICENSE HEAD or we need to move beehive out of kubeedge and use its own LICENSE.
> 
> @qizha thoughts?

I don't see any reason to move it out, so let's keep it.

**Comment by qizha** [comment]
LGTM

File: vendor/github.com/ServiceComb/go-archaius/.gitignore (hunk 7 above)
Comment (m1093782566) [reply vendor/github.com/ServiceComb/go-archaius/.gitignore:15]: Okay, no problem.

**Review by m1093782566 — APPROVED** [review]
LGTM

We can add the LICENSE in follow-ups.

**Pull request merged**